kanidm_tools/src/cli/session.rs

```diff
@@ -142,6 +142,18 @@ impl LoginOpt {
         client.auth_step_password(password.as_str())
     }
 
+    fn do_backup_code(&self, client: &mut KanidmClient) -> Result<AuthResponse, ClientError> {
+        print!("Enter Backup Code: ");
+        // We flush stdout so it'll write the buffer to screen, continuing operation. Without it, the application halts.
+        io::stdout().flush().unwrap();
+        let mut backup_code = String::new();
+        if let Err(e) = io::stdin().read_line(&mut backup_code) {
+            eprintln!("Failed to read from stdin -> {:?}", e);
+            return Err(ClientError::SystemError);
+        };
```

Comment:
Is it worth a check that backup_code.trim().len() > 0, so that we don't post empty strings where someone just accidentally hit return?